### PR TITLE
Add containerized option to kubelet binary

### PR DESCRIPTION
**Pull request by pmorie**
@vmarmol PTAL

### CODE REVIEW

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -213,6 +215,7 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	// Flags intended for testing, not recommended used in production environments.
 	fs.BoolVar(&s.ReallyCrashForTesting, "really-crash-for-testing", s.ReallyCrashForTesting, "If true, when panics occur crash. Intended for testing.")
 	fs.Float64Var(&s.ChaosChance, "chaos-chance", s.ChaosChance, "If > 0.0, introduce random client errors and latency. Intended for testing. [default=0.0]")
+	fs.BoolVar(&s.Containerized, "containerized", s.Containerized, "Experimental support for running kubelet in a container.  Intended for testing.")
```
Comment (vmarmol) [inline cmd/kubelet/app/server.go]: nit: add "Default: false"

**Comment by vmarmol** [comment]
LGTM

**Comment by pmorie** [comment]
@vmarmol Fixed

**Comment by vmarmol** [comment]
Thanks @pmorie! Will merge on green.

**Comment by pmorie** [comment]
@vmarmol tomorrow I'll do PRs for the image and local-up-cluster accommodation for this use-case

**Comment by vmarmol** [comment]
Awesome! Merged :)

**Comment by ixdy** [comment]
This PR broke the non-linux builds:

```
+++ [0504 23:19:57] darwin/amd64: go build started
# github.com/GoogleCloudPlatform/kubernetes/cmd/kubelet/app
_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/cmd/kubelet/app/server.go:280: undefined: mount.NsenterMounter
!!! Error in /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:340
  'go install "${goflags[@]:+${goflags[@]}}" -ldflags "${version_ldflags}" "${nonstatics[@]:+${nonstatics[@]}}"' exited with status 2
Call stack:
  1: /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:340 kube::golang::build_binaries_for_platform(...)
  2: /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:446 kube::golang::build_binaries(...)
  3: hack/build-cross.sh:31 main(...)
Exiting with status 1
+++ [0504 23:19:57] darwin/386: go build started
# github.com/GoogleCloudPlatform/kubernetes/cmd/kubelet/app
_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/cmd/kubelet/app/server.go:280: undefined: mount.NsenterMounter
!!! Error in /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:340
  'go install "${goflags[@]:+${goflags[@]}}" -ldflags "${version_ldflags}" "${nonstatics[@]:+${nonstatics[@]}}"' exited with status 2
Call stack:
  1: /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:340 kube::golang::build_binaries_for_platform(...)
  2: /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:446 kube::golang::build_binaries(...)
  3: hack/build-cross.sh:31 main(...)
Exiting with status 1
+++ [0504 23:19:57] windows/amd64: go build started
# github.com/GoogleCloudPlatform/kubernetes/cmd/kubelet/app
_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/cmd/kubelet/app/server.go:280: undefined: mount.NsenterMounter
!!! Error in /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:340
  'go install "${goflags[@]:+${goflags[@]}}" -ldflags "${version_ldflags}" "${nonstatics[@]:+${nonstatics[@]}}"' exited with status 2
Call stack:
  1: /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:340 kube::golang::build_binaries_for_platform(...)
  2: /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:446 kube::golang::build_binaries(...)
  3: hack/build-cross.sh:31 main(...)
Exiting with status 1
```

**Comment by vmarmol** [comment]
Grr sorry @ixdy. @pmorie we need to add the stub version for non-Linux systems.

Will revert and we'll try again. Thanks for the notice!

**Comment by pmorie** [comment]
Dang

**Comment by pmorie** [comment]
My bad! I will fix

**Comment by vmarmol** [comment]
No worries, should have caught it in review. Also, we're not running build cross in CI? Let me look into fixing that.

**Pull request merged**